kafka-connect-http/src/main/scala/io/lenses/streamreactor/connect/http/sink/tpl/binding/KafkaConnectKeyBinding.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.http.sink.tpl.binding
+
+import cats.implicits.toBifunctorOps
+import io.lenses.streamreactor.connect.cloud.common.sink.extractors.KafkaConnectExtractor
+import org.apache.kafka.connect.errors.ConnectException
+import org.apache.kafka.connect.sink.SinkRecord
+
+class KafkaConnectKeyBinding(name: Option[String]) extends KafkaConnectBaseBinding() {
+  override def get(sinkRecord: SinkRecord): AnyRef = KafkaConnectExtractor.extractFromKey(sinkRecord, name).leftMap(e =>
+    throw new ConnectException(s"unable to extract field $name for template, ", e),
```

Comment:
It's a nit pick :) 
I would simply pattern match on the result of `extractFromKey` and throw on the left branch. It's a bit more explicit behavior wise